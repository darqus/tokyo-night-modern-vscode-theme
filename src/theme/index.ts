export * from './types';
export * from './palette';
export * from './utils';
export * from './generator';
