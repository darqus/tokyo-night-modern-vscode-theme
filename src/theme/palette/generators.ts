/**
 * Функции генерации цветов на основе базовой палитры
 * Реализует механизмы изменения яркости и прозрачности
 */

/**
 * Преобразует HEX цвет в HSL
 */
function hexToHsl(hex: string): { h: number; s: number; l: number } {
  // Удаление символа # и разбиение на компоненты
  const cleanHex = hex.replace('#', '')
  const r = parseInt(cleanHex.substring(0, 2), 16) / 255
  const g = parseInt(cleanHex.substring(2, 4), 16) / 255
  const b = parseInt(cleanHex.substring(4, 6), 16) / 255

  const max = Math.max(r, g, b)
  const min = Math.min(r, g, b)
  let h = 0
  let s = 0
  const l = (max + min) / 2

  if (max === min) {
    h = s = 0 // achromatic
  } else {
    const d = max - min
    s = l > 0.5 ? d / (2 - max - min) : d / (max + min)

    switch (max) {
      case r:
        h = (g - b) / d + (g < b ? 6 : 0)
        break
      case g:
        h = (b - r) / d + 2
        break
      case b:
        h = (r - g) / d + 4
        break
    }

    h /= 6
  }

  return {
    h: Math.round(h * 360),
    s: Math.round(s * 100),
    l: Math.round(l * 100),
  }
}

/**
 * Преобразует HSL цвет в HEX
 */
function hslToHex(h: number, s: number, l: number): string {
  h = h / 360
  s = s / 100
  l = l / 100

  const hue2rgb = (p: number, q: number, t: number): number => {
    if (t < 0) t += 1
    if (t > 1) t -= 1
    if (t < 1 / 6) return p + (q - p) * 6 * t
    if (t < 1 / 2) return q
    if (t < 2 / 3) return p + (q - p) * (2 / 3 - t) * 6
    return p
  }

  let r: number, g: number, b: number

  if (s === 0) {
    r = g = b = l // achromatic
  } else {
    const q = l < 0.5 ? l * (1 + s) : l + s - l * s
    const p = 2 * l - q
    r = hue2rgb(p, q, h + 1 / 3)
    g = hue2rgb(p, q, h)
    b = hue2rgb(p, q, h - 1 / 3)
  }

  const toHex = (c: number): string => {
    const hex = Math.round(c * 255).toString(16)
    return hex.length === 1 ? '0' + hex : hex
  }

  return `#${toHex(r)}${toHex(g)}${toHex(b)}`
}

/**
 * Изменяет яркость цвета
 * @param color - исходный цвет в формате HEX
 * @param amount - величина изменения яркости от -1 до 1 (0.1 = +10% яркости)
 */
export function adjustBrightness(color: string, amount: number): string {
  const hsl = hexToHsl(color)
  let newL = hsl.l + amount * 100
  newL = Math.max(0, Math.min(100, newL)) // Ограничение от 0 до 100
  return hslToHex(hsl.h, hsl.s, newL)
}

/**
 * Делает цвет светлее
 * @param color - исходный цвет в формате HEX
 * @param amount - величина осветления от 0 до 1 (0.1 = +10% яркости)
 */
export function lighten(color: string, amount: number): string {
  return adjustBrightness(color, Math.abs(amount))
}

/**
 * Делает цвет темнее
 * @param color - исходный цвет в формате HEX
 * @param amount - величина затемнения от 0 до 1 (0.1 = -10% яркости)
 */
export function darken(color: string, amount: number): string {
  return adjustBrightness(color, -Math.abs(amount))
}

/**
 * Добавляет прозрачность к цвету
 * @param color - исходный цвет в формате HEX
 * @param alpha - значение прозрачности от 0 до 1 (0.5 = 50% прозрачности)
 */
export function withAlpha(color: string, alpha: number): string {
  const cleanHex = color.replace('#', '')
  const alphaHex = Math.round(alpha * 255)
    .toString(16)
    .padStart(2, '0')
  return `#${cleanHex}${alphaHex}`
}

/**
 * Интерфейс для системы прозрачности
 */
export interface AlphaSystem {
  subtle: string // 20% (33)
  hover: string // 40% (66)
  active: string // 60% (99)
  semiOpaque: string // 80% (cc)
}

/**
 * Создает систему прозрачности на основе значений в шестнадцатеричном формате
 */
export function createAlphaSystem(): AlphaSystem {
  return {
    subtle: '33', // 20% прозрачности
    hover: '66', // 40% прозрачности
    active: '99', // 60% прозрачности
    semiOpaque: 'cc', // 80% прозрачности
  }
}

/**
 * Возвращает цвет с заданным уровнем прозрачности из системы
 * @param color - исходный цвет в формате HEX
 * @param alphaKey - ключ уровня прозрачности
 * @param alphaSystem - система прозрачности
 */
export function getColorWithAlpha(
  color: string,
  alphaKey: keyof AlphaSystem,
  alphaSystem: AlphaSystem = createAlphaSystem()
): string {
  const hex = color.replace('#', '')
  return `#${hex}${alphaSystem[alphaKey]}`
}
