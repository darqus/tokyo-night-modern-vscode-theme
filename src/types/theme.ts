import type { Hex } from './palette'

/**
 * Основные типы для улучшенной архитектуры темы
 */

export interface ThemeVariant {
  id: string
  name: string
  displayName: string
  baseConfig: ThemeConfig
}

export interface ThemeConfig {
  contrast: 'low' | 'normal' | 'high'
  saturation: 'muted' | 'normal' | 'vibrant'
  brightness: 'darker' | 'normal' | 'lighter'
}

export interface ComponentStates {
  default: ComponentStyle
  hover?: ComponentStyle
  active?: ComponentStyle
  focus?: ComponentStyle
  disabled?: ComponentStyle
}

export interface ComponentStyle {
  background?: Hex | 'transparent'
  foreground?: Hex
  border?: Hex | 'transparent'
  shadow?: string
}

export interface UIComponent {
  name: string
  states: ComponentStates
  variants?: ComponentVariants
}

export interface ComponentVariants {
  [key: string]: ComponentStates
}

export interface ColorScale {
  50: Hex // Самый светлый
  100: Hex
  200: Hex
  300: Hex
  400: Hex
  500: Hex // Базовый
  600: Hex
  700: Hex
  800: Hex
  900: Hex // Самый темный
}

export interface ValidationRule {
  name: string
  description: string
  validate: (theme: ThemeData) => ValidationResult
}

export interface ValidationResult {
  passed: boolean
  issues: ValidationIssue[]
}

export interface ValidationIssue {
  severity: 'error' | 'warning' | 'info'
  message: string
  suggestion?: string
}

export interface ThemeData {
  name: string
  type: 'dark' | 'light'
  colors: Record<string, string>
  tokenColors: any[]
  semanticTokenColors: Record<string, any>
}

export interface ThemePlugin {
  name: string
  version: string
  description: string
  apply: (theme: ThemeData, config?: any) => ThemeData
  isCompatible: (themeVersion: string) => boolean
}

export interface ThemeSettings {
  variant: string
  contrast: 'low' | 'normal' | 'high'

  syntax: {
    enableSemanticHighlighting: boolean
    bracketPairColorization: boolean
    customLanguageSupport: string[]
  }

  ui: {
    activityBarPosition: 'left' | 'right'
    tabBarStyle: 'classic' | 'modern' | 'minimal'
    statusBarStyle: 'minimal' | 'full'
  }

  customColors?: Record<string, string>
}
