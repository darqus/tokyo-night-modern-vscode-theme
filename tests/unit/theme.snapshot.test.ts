import { generateTheme } from '../../src/generators/theme'

/**
 * Snapshot-тест итоговой темы. Любые изменения палитры/токенов
 * должны отражаться через осознанное обновление снапшота.
 */

describe('Theme snapshot', () => {
  it('should match the stable snapshot of the generated theme', () => {
    const theme = generateTheme()

    // Базовые инварианты
    expect(theme.name).toBe('Tokyo Night Modern')
    expect(theme.type).toBe('dark')
    expect(theme.colors).toBeDefined()
    expect(theme.tokenColors).toBeDefined()

    // Снапшот всей структуры
    expect(theme).toMatchSnapshot()
  })
})
