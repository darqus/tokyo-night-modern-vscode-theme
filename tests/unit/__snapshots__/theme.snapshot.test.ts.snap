// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme snapshot should match the stable snapshot of the generated theme 1`] = `
{
  "colors": {
    "activityBar.activeBackground": "#7aa2f714",
    "activityBar.activeBorder": "#7dcfff",
    "activityBar.background": "#1a1b26",
    "activityBar.border": "#2c2f44",
    "activityBar.foreground": "#969fca",
    "activityBar.inactiveForeground": "#6b749f",
    "activityBarBadge.background": "#7dcfff",
    "activityBarBadge.foreground": "#1a1b26",
    "badge.background": "#7dcfff",
    "badge.foreground": "#1a1b26",
    "breadcrumb.activeSelectionForeground": "#c0caf5",
    "breadcrumb.background": "#1a1b26",
    "breadcrumb.focusForeground": "#c0caf5",
    "breadcrumb.foreground": "#969fca",
    "breadcrumbPicker.background": "#202333",
    "button.background": "#2d3650",
    "button.border": "#2c2f44",
    "button.foreground": "#c0caf5",
    "button.hoverBackground": "#323d5a",
    "button.secondaryBackground": "#24293b",
    "button.secondaryForeground": "#c0caf5",
    "button.secondaryHoverBackground": "#7aa2f714",
    "button.separator": "#2c2f44",
    "charts.blue": "#7dcfff",
    "charts.foreground": "#c0caf5",
    "charts.green": "#9ece6a",
    "charts.lines": "#2c2f44",
    "charts.orange": "#e0af68",
    "charts.purple": "#7dcfff",
    "charts.red": "#f7768e",
    "charts.yellow": "#e0af68",
    "checkbox.background": "#202333",
    "checkbox.border": "#2c2f44",
    "checkbox.foreground": "#c0caf5",
    "commandCenter.activeBackground": "#7aa2f714",
    "commandCenter.activeForeground": "#c0caf5",
    "commandCenter.background": "#1a1b26",
    "commandCenter.border": "#2c2f44",
    "commandCenter.foreground": "#969fca",
    "debugConsole.errorForeground": "#f7768e",
    "debugConsole.infoForeground": "#7dcfff",
    "debugConsole.sourceForeground": "#c0caf5",
    "debugConsole.warningForeground": "#e0af68",
    "debugConsoleInputIcon.foreground": "#c0caf5",
    "debugExceptionWidget.background": "#f7768e",
    "debugExceptionWidget.border": "#f7768e",
    "debugIcon.breakpointCurrentStackframeForeground": "#e0af68",
    "debugIcon.breakpointDisabledForeground": "#6b749f",
    "debugIcon.breakpointForeground": "#f7768e",
    "debugIcon.breakpointStackframeForeground": "#7dcfff",
    "debugIcon.breakpointUnverifiedForeground": "#969fca",
    "debugIcon.continueForeground": "#9ece6a",
    "debugIcon.disconnectForeground": "#f7768e",
    "debugIcon.pauseForeground": "#7dcfff",
    "debugIcon.restartForeground": "#9ece6a",
    "debugIcon.startForeground": "#9ece6a",
    "debugIcon.stepBackForeground": "#7dcfff",
    "debugIcon.stepIntoForeground": "#7dcfff",
    "debugIcon.stepOutForeground": "#7dcfff",
    "debugIcon.stepOverForeground": "#7dcfff",
    "debugIcon.stopForeground": "#f7768e",
    "debugToolBar.background": "#202333",
    "debugToolBar.border": "#2c2f44",
    "descriptionForeground": "#969fca",
    "diffEditor.diagonalFill": "#24293b",
    "diffEditor.insertedLineBackground": "#9ece6a1a",
    "diffEditor.insertedTextBackground": "#9ece6a26",
    "diffEditor.removedLineBackground": "#f7768e1a",
    "diffEditor.removedTextBackground": "#f7768e26",
    "disabledForeground": "#6b749f",
    "dropdown.background": "#202333",
    "dropdown.border": "#2c2f44",
    "dropdown.foreground": "#c0caf5",
    "dropdown.listBackground": "#202333",
    "editor.background": "#1a1b26",
    "editor.findMatchBackground": "#e0af6838",
    "editor.findMatchBorder": "#e0af68b3",
    "editor.findMatchForeground": "#c0caf5",
    "editor.findMatchHighlightBackground": "#e0af6824",
    "editor.findMatchHighlightForeground": "#c0caf5",
    "editor.foreground": "#c0caf5",
    "editor.hoverHighlightBackground": "#7aa2f714",
    "editor.inactiveSelectionBackground": "#7aa2f714",
    "editor.lineHighlightBackground": "#7aa2f714",
    "editor.rangeHighlightBackground": "#7aa2f714",
    "editor.selectionBackground": "#7aa2f733",
    "editor.selectionForeground": "#c0caf5",
    "editor.wordHighlightBackground": "#7aa2f714",
    "editor.wordHighlightStrongBackground": "#7aa2f733",
    "editorBracketMatch.background": "#7aa2f714",
    "editorBracketMatch.border": "#7aa2f766",
    "editorCursor.foreground": "#c0caf5",
    "editorGroup.border": "#2c2f44",
    "editorGroup.dropBackground": "#24293b33",
    "editorGroupHeader.noTabsBackground": "#1a1b26",
    "editorGroupHeader.tabsBackground": "#1a1b26",
    "editorGroupHeader.tabsBorder": "#2c2f44",
    "editorHoverWidget.background": "#202333",
    "editorHoverWidget.border": "#2c2f44",
    "editorHoverWidget.foreground": "#c0caf5",
    "editorHoverWidget.highlightForeground": "#7dcfff",
    "editorHoverWidget.statusBarBackground": "#24293b",
    "editorIndentGuide.activeBackground1": "#7aa2f766",
    "editorIndentGuide.background1": "#2c2f44",
    "editorLineNumber.activeForeground": "#726f82",
    "editorLineNumber.foreground": "#3e4461",
    "editorMarkerNavigation.background": "#202333",
    "editorMarkerNavigationError.background": "#f7768e",
    "editorMarkerNavigationInfo.background": "#7dcfff",
    "editorMarkerNavigationWarning.background": "#e0af68",
    "editorOverviewRuler.addedForeground": "#9ece6a",
    "editorOverviewRuler.border": "#2c2f44",
    "editorOverviewRuler.bracketMatchForeground": "#7aa2f766",
    "editorOverviewRuler.deletedForeground": "#f7768e",
    "editorOverviewRuler.errorForeground": "#f7768e",
    "editorOverviewRuler.findMatchForeground": "#e0af68",
    "editorOverviewRuler.infoForeground": "#7dcfff",
    "editorOverviewRuler.modifiedForeground": "#7dcfff",
    "editorOverviewRuler.rangeHighlightForeground": "#7dcfff",
    "editorOverviewRuler.selectionHighlightForeground": "#7aa2f733",
    "editorOverviewRuler.warningForeground": "#e0af68",
    "editorOverviewRuler.wordHighlightForeground": "#7aa2f714",
    "editorOverviewRuler.wordHighlightStrongForeground": "#7aa2f733",
    "editorRuler.foreground": "#2c2f44",
    "editorSuggestWidget.background": "#202333",
    "editorSuggestWidget.border": "#2c2f44",
    "editorSuggestWidget.foreground": "#c0caf5",
    "editorSuggestWidget.highlightForeground": "#7dcfff",
    "editorSuggestWidget.selectedBackground": "#7aa2f733",
    "editorSuggestWidget.selectedForeground": "#c0caf5",
    "editorSuggestWidget.selectedIconForeground": "#c0caf5",
    "editorSuggestWidgetStatus.foreground": "#969fca",
    "editorWhitespace.foreground": "#808ab4",
    "editorWidget.background": "#202333",
    "editorWidget.border": "#2c2f44",
    "editorWidget.foreground": "#c0caf5",
    "editorWidget.resizeBorder": "#7aa2f766",
    "errorForeground": "#f7768e",
    "extensionBadge.remoteBackground": "#7dcfff",
    "extensionBadge.remoteForeground": "#1a1b26",
    "extensionButton.prominentBackground": "#7dcfff",
    "extensionButton.prominentForeground": "#1a1b26",
    "extensionButton.prominentHoverBackground": "#7dcfff",
    "focusBorder": "#7aa2f766",
    "foreground": "#c0caf5",
    "gitDecoration.addedResourceForeground": "#9ece6a",
    "gitDecoration.conflictingResourceForeground": "#e0af68",
    "gitDecoration.deletedResourceForeground": "#f7768e",
    "gitDecoration.ignoredResourceForeground": "#6b749f",
    "gitDecoration.modifiedResourceForeground": "#7dcfff",
    "gitDecoration.renamedResourceForeground": "#7dcfffcc",
    "gitDecoration.stageDeletedResourceForeground": "#f7768ee6",
    "gitDecoration.stageModifiedResourceForeground": "#7dcfffe6",
    "gitDecoration.submoduleResourceForeground": "#969fca",
    "gitDecoration.untrackedResourceForeground": "#9ece6a",
    "icon.foreground": "#969fca",
    "inlineChat.background": "#202333",
    "inlineChat.border": "#2c2f44",
    "inlineChat.foreground": "#c0caf5",
    "input.background": "#1e202e",
    "input.border": "#2c2f44",
    "input.foreground": "#c0caf5",
    "input.placeholderForeground": "#808ab4",
    "inputOption.activeBackground": "#7aa2f733",
    "inputOption.activeBorder": "#7aa2f766",
    "inputValidation.errorBackground": "#f7768e",
    "inputValidation.errorBorder": "#f7768e",
    "inputValidation.errorForeground": "#1a1b26",
    "inputValidation.infoBackground": "#7dcfff",
    "inputValidation.infoBorder": "#7dcfff",
    "inputValidation.infoForeground": "#1a1b26",
    "inputValidation.warningBackground": "#e0af68",
    "inputValidation.warningBorder": "#e0af68",
    "inputValidation.warningForeground": "#1a1b26",
    "keybindingLabel.background": "#202333",
    "keybindingLabel.border": "#2c2f44",
    "keybindingLabel.bottomBorder": "#2c2f44",
    "keybindingLabel.foreground": "#c0caf5",
    "list.activeSelectionBackground": "#7aa2f733",
    "list.activeSelectionForeground": "#c0caf5",
    "list.dropBackground": "#24293b33",
    "list.errorForeground": "#f7768e",
    "list.filterMatchBackground": "#e0af6824",
    "list.filterMatchBorder": "#e0af68b3",
    "list.focusBackground": "#7aa2f733",
    "list.focusForeground": "#c0caf5",
    "list.highlightForeground": "#7dcfff",
    "list.hoverBackground": "#7aa2f714",
    "list.hoverForeground": "#c0caf5",
    "list.inactiveSelectionBackground": "#7aa2f714",
    "list.inactiveSelectionForeground": "#c0caf5",
    "list.invalidItemForeground": "#f7768e",
    "list.warningForeground": "#e0af68",
    "menu.background": "#202333",
    "menu.border": "#2c2f44",
    "menu.foreground": "#c0caf5",
    "menu.selectionBackground": "#7aa2f733",
    "menu.selectionBorder": "#7aa2f766",
    "menu.selectionForeground": "#c0caf5",
    "menu.separatorBackground": "#2c2f44",
    "menubar.selectionBackground": "#7aa2f714",
    "menubar.selectionBorder": "#7aa2f766",
    "menubar.selectionForeground": "#c0caf5",
    "merge.border": "#2c2f44",
    "minimap.background": "#1a1b26",
    "minimap.errorHighlight": "#f7768e",
    "minimap.findMatchHighlight": "#e0af6880",
    "minimap.selectionHighlight": "#7aa2f733",
    "minimap.selectionOccurrenceHighlight": "#7aa2f714",
    "minimap.warningHighlight": "#e0af68",
    "minimapGutter.addedBackground": "#9ece6a",
    "minimapGutter.deletedBackground": "#f7768e",
    "minimapGutter.modifiedBackground": "#7dcfff",
    "minimapSlider.activeBackground": "#7aa2f733",
    "minimapSlider.background": "#7aa2f714",
    "minimapSlider.hoverBackground": "#7aa2f71f",
    "notificationCenter.border": "#2c2f44",
    "notificationCenterHeader.background": "#202333",
    "notificationCenterHeader.foreground": "#c0caf5",
    "notificationToast.border": "#2c2f44",
    "notifications.background": "#202333",
    "notifications.border": "#2c2f44",
    "notifications.foreground": "#c0caf5",
    "notificationsErrorIcon.foreground": "#f7768e",
    "notificationsInfoIcon.foreground": "#7dcfff",
    "notificationsWarningIcon.foreground": "#e0af68",
    "panel.background": "#1a1b26",
    "panel.border": "#2c2f44",
    "panel.dropBorder": "#7dcfff",
    "panelInput.border": "#2c2f44",
    "panelSection.border": "#2c2f44",
    "panelSection.dropBackground": "#24293b33",
    "panelSectionHeader.background": "#202333",
    "panelSectionHeader.border": "#2c2f44",
    "panelSectionHeader.foreground": "#c0caf5",
    "panelTitle.activeBorder": "#7dcfff",
    "panelTitle.activeForeground": "#c0caf5",
    "panelTitle.inactiveForeground": "#969fca",
    "peekView.border": "#7aa2f766",
    "peekViewEditor.background": "#202333",
    "peekViewEditor.matchHighlightBackground": "#7aa2f733",
    "peekViewResult.background": "#1a1b26",
    "peekViewResult.fileForeground": "#c0caf5",
    "peekViewResult.lineForeground": "#969fca",
    "peekViewResult.matchHighlightBackground": "#7aa2f733",
    "peekViewResult.selectionBackground": "#7aa2f733",
    "peekViewResult.selectionForeground": "#c0caf5",
    "peekViewTitle.background": "#24293b",
    "peekViewTitleDescription.foreground": "#969fca",
    "peekViewTitleLabel.foreground": "#c0caf5",
    "ports.iconRunningProcessForeground": "#9ece6a",
    "problemsErrorIcon.foreground": "#f7768e",
    "problemsInfoIcon.foreground": "#7dcfff",
    "problemsWarningIcon.foreground": "#e0af68",
    "progressBar.background": "#7dcfff",
    "quickInput.background": "#202333",
    "quickInput.foreground": "#c0caf5",
    "quickInputList.focusBackground": "#7aa2f733",
    "quickInputList.focusForeground": "#c0caf5",
    "quickInputTitle.background": "#24293b",
    "scmGraph.foreground1": "#7aa2f7",
    "scmGraph.foreground2": "#bb9af7",
    "scmGraph.foreground3": "#73daca",
    "scmGraph.foreground4": "#7dcfff",
    "scmGraph.foreground5": "#9d7cd8",
    "scmGraph.historyItemBaseRefColor": "#73daca",
    "scmGraph.historyItemHoverAdditionsForeground": "#9ece6a",
    "scmGraph.historyItemHoverDefaultLabelBackground": "#7aa2f7",
    "scmGraph.historyItemHoverDefaultLabelForeground": "#1a1b26",
    "scmGraph.historyItemHoverDeletionsForeground": "#f7768e",
    "scmGraph.historyItemHoverLabelForeground": "#1a1b26",
    "scmGraph.historyItemRefColor": "#7aa2f7",
    "scmGraph.historyItemRemoteRefColor": "#bb9af7",
    "scrollbar.shadow": "#24293b",
    "scrollbarSlider.activeBackground": "#7aa2f733",
    "scrollbarSlider.background": "#7aa2f714",
    "scrollbarSlider.hoverBackground": "#7aa2f71f",
    "searchEditor.findMatchBackground": "#e0af6824",
    "searchEditor.findMatchBorder": "#e0af68b3",
    "selection.background": "#7aa2f733",
    "settings.checkboxBackground": "#202333",
    "settings.checkboxBorder": "#2c2f44",
    "settings.checkboxForeground": "#c0caf5",
    "settings.dropdownBackground": "#202333",
    "settings.dropdownBorder": "#2c2f44",
    "settings.dropdownForeground": "#c0caf5",
    "settings.headerForeground": "#c0caf5",
    "settings.modifiedItemIndicator": "#7dcfff",
    "settings.numberInputBackground": "#1e202e",
    "settings.numberInputBorder": "#2c2f44",
    "settings.numberInputForeground": "#c0caf5",
    "settings.textInputBackground": "#1e202e",
    "settings.textInputBorder": "#2c2f44",
    "settings.textInputForeground": "#c0caf5",
    "sideBar.background": "#1a1b26",
    "sideBar.border": "#2c2f44",
    "sideBar.foreground": "#c0caf5",
    "sideBarSectionHeader.background": "#202333",
    "sideBarSectionHeader.border": "#2c2f44",
    "sideBarSectionHeader.foreground": "#c0caf5",
    "sideBarTitle.foreground": "#c0caf5",
    "statusBar.background": "#1a1b26",
    "statusBar.border": "#2c2f44",
    "statusBar.debuggingBackground": "#e0af68",
    "statusBar.debuggingForeground": "#1a1b26",
    "statusBar.foreground": "#969fca",
    "statusBar.noFolderBackground": "#1a1b26",
    "statusBarItem.activeBackground": "#969fca26",
    "statusBarItem.compactHoverBackground": "#969fca14",
    "statusBarItem.errorBackground": "#f7768e",
    "statusBarItem.errorForeground": "#1a1b26",
    "statusBarItem.errorHoverBackground": "#f7768ecc",
    "statusBarItem.errorHoverForeground": "#1a1b26",
    "statusBarItem.hoverBackground": "#969fca1f",
    "statusBarItem.hoverForeground": "#c0caf5",
    "statusBarItem.prominentBackground": "#7dcfff",
    "statusBarItem.prominentForeground": "#1a1b26",
    "statusBarItem.prominentHoverBackground": "#7dcfffcc",
    "statusBarItem.prominentHoverForeground": "#1a1b26",
    "statusBarItem.remoteBackground": "#7dcfff",
    "statusBarItem.remoteForeground": "#1a1b26",
    "statusBarItem.remoteHoverBackground": "#7dcfffcc",
    "statusBarItem.remoteHoverForeground": "#1a1b26",
    "statusBarItem.warningBackground": "#e0af68",
    "statusBarItem.warningForeground": "#1a1b26",
    "statusBarItem.warningHoverBackground": "#e0af68cc",
    "statusBarItem.warningHoverForeground": "#1a1b26",
    "tab.activeBackground": "#202333",
    "tab.activeBorderTop": "#7dcfff",
    "tab.activeForeground": "#c0caf5",
    "tab.border": "#2c2f44",
    "tab.hoverBackground": "#7aa2f714",
    "tab.hoverForeground": "#c0caf5",
    "tab.inactiveBackground": "#1a1b26",
    "tab.inactiveForeground": "#969fca",
    "tab.lastPinnedBorder": "#2c2f44",
    "tab.unfocusedActiveBackground": "#202333",
    "tab.unfocusedActiveForeground": "#969fca",
    "tab.unfocusedInactiveBackground": "#1a1b26",
    "tab.unfocusedInactiveForeground": "#6b749f",
    "terminal.ansiBlack": "#1a1b26",
    "terminal.ansiBlue": "#7aa2f7",
    "terminal.ansiBrightBlack": "#76767d",
    "terminal.ansiBrightBlue": "#8eb0f8",
    "terminal.ansiBrightCyan": "#8ad4ff",
    "terminal.ansiBrightGreen": "#add580",
    "terminal.ansiBrightMagenta": "#c5a9f8",
    "terminal.ansiBrightRed": "#f88b9f",
    "terminal.ansiBrightWhite": "#c6cff6",
    "terminal.ansiBrightYellow": "#e5bb7f",
    "terminal.ansiCyan": "#7dcfff",
    "terminal.ansiGreen": "#9ece6a",
    "terminal.ansiMagenta": "#bb9af7",
    "terminal.ansiRed": "#f7768e",
    "terminal.ansiWhite": "#c0caf5",
    "terminal.ansiYellow": "#e0af68",
    "terminal.background": "#1a1b26",
    "terminal.border": "#2c2f44",
    "terminal.dropBackground": "#24293b33",
    "terminal.findMatchBackground": "#e0af6838",
    "terminal.findMatchBorder": "#e0af68b3",
    "terminal.findMatchHighlightBackground": "#e0af6824",
    "terminal.foreground": "#c0caf5",
    "terminal.hoverHighlightBackground": "#7aa2f714",
    "terminal.inactiveSelectionBackground": "#7aa2f714",
    "terminal.initialHintForeground": "#969fca",
    "terminal.selectionBackground": "#7aa2f733",
    "terminal.tab.activeBorder": "#7dcfff",
    "terminalCommandDecoration.defaultBackground": "#969fca1f",
    "terminalCommandDecoration.errorBackground": "#f7768e",
    "terminalCommandDecoration.successBackground": "#9ece6a",
    "terminalCursor.background": "#1a1b26",
    "terminalCursor.foreground": "#c0caf5",
    "terminalOverviewRuler.border": "#2c2f44",
    "terminalOverviewRuler.cursorForeground": "#7dcfff",
    "terminalOverviewRuler.findMatchForeground": "#e0af68",
    "testing.iconErrored": "#f7768e",
    "testing.iconFailed": "#f7768e",
    "testing.iconPassed": "#9ece6a",
    "testing.iconQueued": "#e0af68",
    "testing.iconSkipped": "#6b749f",
    "testing.iconUnset": "#969fca",
    "testing.peekBorder": "#7aa2f766",
    "testing.peekHeaderBackground": "#24293b",
    "testing.runAction": "#9ece6a",
    "textLink.activeForeground": "#8ad4ff",
    "textLink.foreground": "#7dcfff",
    "titleBar.activeBackground": "#1a1b26",
    "titleBar.activeForeground": "#c0caf5",
    "titleBar.border": "#2c2f44",
    "titleBar.inactiveBackground": "#1a1b26",
    "titleBar.inactiveForeground": "#969fca",
    "toolbar.activeBackground": "#7aa2f733",
    "toolbar.hoverBackground": "#7aa2f714",
    "toolbar.hoverOutline": "#7aa2f766",
    "tree.indentGuidesStroke": "#2c2f44",
    "walkThrough.embeddedEditorBackground": "#202333",
    "welcomePage.background": "#1a1b26",
    "welcomePage.progress.background": "#202333",
    "welcomePage.progress.foreground": "#7dcfff",
    "welcomePage.tileBackground": "#202333",
    "welcomePage.tileBorder": "#2c2f44",
    "welcomePage.tileHoverBackground": "#7aa2f714",
    "widget.border": "#2c2f44",
    "widget.shadow": "#24293b",
  },
  "name": "Tokyo Night Modern",
  "semanticTokenColors": {
    "class": {
      "foreground": "#e0af68",
    },
    "comment": {
      "foreground": "#565f89",
      "italic": true,
    },
    "deprecated": {
      "foreground": "#767fa9",
      "strikethrough": true,
    },
    "function": {
      "foreground": "#7aa2f7",
    },
    "interface": {
      "foreground": "#73daca",
    },
    "keyword": {
      "foreground": "#bb9af7",
    },
    "method": {
      "foreground": "#7aa2f7",
    },
    "number": {
      "foreground": "#ff9e64",
    },
    "parameter": {
      "foreground": "#7dcfff",
    },
    "property": {
      "foreground": "#7dcfff",
    },
    "string": {
      "foreground": "#9ece6a",
    },
    "type": {
      "foreground": "#73daca",
    },
    "variable": {
      "foreground": "#7dcfff",
    },
    "variable.readonly": {
      "foreground": "#e0af68",
    },
  },
  "tokenColors": [
    {
      "name": "Comment",
      "scope": [
        "comment",
        "punctuation.definition.comment",
      ],
      "settings": {
        "fontStyle": "italic",
        "foreground": "#565f89",
      },
    },
    {
      "name": "Keyword",
      "scope": [
        "keyword",
        "storage.type",
        "storage.modifier",
      ],
      "settings": {
        "foreground": "#bb9af7",
      },
    },
    {
      "name": "String",
      "scope": [
        "string",
        "constant.other.symbol",
      ],
      "settings": {
        "foreground": "#9ece6a",
      },
    },
    {
      "name": "Number",
      "scope": [
        "constant.numeric",
        "constant.language",
      ],
      "settings": {
        "foreground": "#ff9e64",
      },
    },
    {
      "name": "Function",
      "scope": [
        "entity.name.function",
        "meta.function-call",
        "variable.function",
      ],
      "settings": {
        "foreground": "#7aa2f7",
      },
    },
    {
      "name": "Variable",
      "scope": [
        "variable",
        "support.variable",
      ],
      "settings": {
        "foreground": "#7dcfff",
      },
    },
    {
      "name": "Class",
      "scope": [
        "entity.name.class",
        "entity.name.type",
        "support.class",
      ],
      "settings": {
        "foreground": "#e0af68",
      },
    },
    {
      "name": "Property",
      "scope": [
        "variable.other.property",
        "meta.object-literal.key",
      ],
      "settings": {
        "foreground": "#7dcfff",
      },
    },
    {
      "name": "Tag",
      "scope": [
        "entity.name.tag",
      ],
      "settings": {
        "foreground": "#f7768e",
      },
    },
    {
      "name": "Attribute",
      "scope": [
        "entity.other.attribute-name",
      ],
      "settings": {
        "foreground": "#bb9af7",
      },
    },
    {
      "name": "Operator",
      "scope": [
        "keyword.operator",
      ],
      "settings": {
        "foreground": "#9d7cd8",
      },
    },
    {
      "name": "Punctuation",
      "scope": [
        "punctuation",
      ],
      "settings": {
        "foreground": "#c0caf5",
      },
    },
    {
      "name": "Invalid",
      "scope": [
        "invalid",
        "invalid.illegal",
      ],
      "settings": {
        "foreground": "#f7768e",
      },
    },
  ],
  "type": "dark",
}
`;
